package/MDAnalysis/analysis/base.py

```diff
@@ -430,25 +471,241 @@ def run(self, start=None, stop=None, step=None, frames=None,
+            and `n_workers` with `dask` scheduler.
+
+        .. versionchanged:: 2.2.0
+            Added ability to analyze arbitrary frames by passing a list of
+            frame indices in the `frames` keyword argument.
+
+        .. versionchanged:: 2.5.0
+            Add `progressbar_kwargs` parameter, 
+            allowing to modify description, position etc of tqdm progressbars
+        """
+        if progressbar_kwargs and scheduler:
+            raise NotImplementedError("Can not display progress if scheduler is not None")
+
+        self._setup_frames(self._trajectory, start, stop, step, frames)
+        self._prepare()
+        self._setup_scheduler(scheduler=scheduler, n_workers=n_workers)
+        self._setup_bslices(start=start, stop=stop, step=step, frames=frames)
+
+
+        if scheduler is None: # fallback to the local scheduler
+            self._compute(bslice_idx=0, verbose=verbose, progressbar_kwargs=progressbar_kwargs)
+        else:
+            if scheduler == 'multiprocessing':
+                import multiprocessing
+
+                delayed = localdelayed
+                computations = [
+                    (self, bslice_idx)
+                    for bslice_idx in range(len(self._bslices))]
+
+                with multiprocessing.Pool(processes=self._scheduler_kwargs['n_workers']) as pool:
+                    results = pool.starmap(_self_compute_helper, computations)
+
+            elif scheduler == 'dask':
+                try:
+                    from dask.delayed import delayed
+                except ImportError:
+                    raise ImportError(
+                        "Please install dask to execute run() in parallel using scheduler='dask'")
+
+                computations = delayed(
+                    [
+                        delayed(self._compute)(bslice_idx=bslice_idx)
+                        for bslice_idx in range(len(self._bslices))
+                    ]
+                )
+                results = computations.compute(**self._scheduler_kwargs)
+
+            else:
+                raise ValueError("scheduler can be 'dask', 'multiprocessing' or None")
+
+            self._remote_results = results
+            self._parallel_conclude()
+
+            self.frames = np.array([obj.frames for obj in self._remote_results]).sum(axis=0)
+            self.times = np.array([obj.times for obj in self._remote_results]).sum(axis=0)
+
         self._conclude()
+
         return self
+    
+    def _parallel_conclude(self):
+        """Aggregation function that gathers together results of workers 
+        into .results of the main object. 
+        
+        By default, if checks type of each attribute in first remote worker,
+        and based on this type applies aggregation to the results in remote workers:
+         - `np.ndarray` -- sums the results (assuming the results will be zero in all but one workers)
```

Comment:
>Does this mean that EACH worker holds FULL arrays for everything
yes, they do, since they're being set up by the `setup_frames` I believe. 
I'll think on how to improve it.

Reply:
@marinegor is this still outstanding?